### PR TITLE
Copy container logs for easier debugging

### PR DESCRIPTION
**Pull request by dims**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_node/remote/remote.go [hunk 1]
```diff
@@ -207,22 +207,31 @@ func getTestArtifacts(host, testDir string) error {
 	if err := os.MkdirAll(logPath, 0755); err != nil {
 		return fmt.Errorf("failed to create log directory %q: %w", logPath, err)
 	}
-	// Copy logs to artifacts/hostname
-	if _, err := runSSHCommand(host, "scp", "-r", fmt.Sprintf("%s:%s/results/*.log", GetHostnameOrIP(host), testDir), logPath); err != nil {
-		return err
+	// Copy logs (if any) to artifacts/hostname
```
Comment (dims) [inline test/e2e_node/remote/remote.go:210]: Check if there are things in `results/*.log` before running scp (just like we do below)

File: test/e2e_node/remote/remote.go [hunk 2]
```diff
@@ -207,22 +207,31 @@ func getTestArtifacts(host, testDir string) error {
...
 	}
+	// Copy junit results (if any) to artifacts
 	if _, err := SSH(host, "ls", fmt.Sprintf("%s/results/junit*", testDir)); err == nil {
 		// Copy junit (if any) to the top of artifacts
 		if _, err = runSSHCommand(host, "scp", fmt.Sprintf("%s:%s/results/junit*", GetHostnameOrIP(host), testDir), *resultsDir); err != nil {
 			return err
 		}
 	}
+	// Copy container logs to artifacts/hostname
+	if _, err := SSH(host, "chmod", "-R", "a+r", "/var/log/pods"); err == nil {
```
Comment (dims) [inline test/e2e_node/remote/remote.go:230]: Ensure the container/pod logs are readable before we run `scp`, if the chmod runs fine, we can just recursively copy over everything below it.

**Review by tzneal — APPROVED** [review]
/lgtm

**Review by upodroid — APPROVED** [review]
/lgtm

**Pull request merged**